src/Matrix.php

```diff
@@ -3860,4 +3964,21 @@ public function __toString() : string
     {
         return array_reduce($this->a, [self::class, 'implodeRow'], '') . PHP_EOL;
     }
+
+    /**
+     * Copies elements from a given matrix
+     *
+     * @param \Tensor\Matrix $b
+     * @param int $i
+     * @param int $j
+     * @return void
+     */
+    protected function copyElements(Matrix $b, int $i, int $j): void
```

Comment:
Double indexing, possible optimization

Reply:
it is not common to do so in PHP world, but yes, PHP follows traditional C-like for loops wherein in all positions you can use multiple expressions. However, while syntactically correct, it is useless to do so for conditionals checks specifically, because first expressions get evaluated and discarded and only the last one is used for the loop-condition.

Reply:
Just when I thought I'd seen everything